### PR TITLE
typescript!!!!

### PR DESCRIPTION
**Pull request by grantoesterling**
## What's New
- Starting to convert Artbot to Typescript!
- I work on this when I'm drained and need something mindless yet productive to do lol
- Pls give some suggestions for the tsconfig - i kinda just copied in random stuff from the website repo one
- Tested pretty decently - everything seems to work pretty well!
- Updating linter to use TS stuff

Converted (so far):
- index, ArtIndexerBot, ProjectBot, APIPollBot, utils, Reservoir List/Sale bots, ArchipelagoBot

Removed (unused code I found):
- LooksRareAPIPollBot
- AddressCollector
- GiveawaysManager

Temporarily removing tests until i can convert them to ts too

TODO post-merge:
- Go back and add proper typing to API responses, plus other "any" types

### CODE REVIEW

**Review by Jbern16 — APPROVED** [review]
wowwweee. I think this is good to go as is knowing its WIP

File: src/Classes/APIBots/ApiPollBot.ts [hunk 1]
```diff
@@ -1,15 +1,23 @@
+import { Client, Message } from 'discord.js'
+
 const { ensOrAddress, getOSName } = require('./utils')
 const axios = require('axios')
 /** Abstract parent class for all API Poll Bots */
 class APIPollBot {
-  /**
```
Comment (asaltste) [inline src/Classes/APIBots/ApiPollBot.ts:5]: nit: I think we should still keep this comment for explanation about instructor

File: src/Classes/APIBots/ReservoirListBot.ts [hunk 2]
```diff
@@ -27,7 +35,7 @@ class ReservoirListBot extends APIPollBot {
    * Response spec: https://docs.reservoir.tools/reference/getordersasksv2
    * @param {*} responseData - Dict parsed from API request json
    */
-  async handleAPIResponse(responseData) {
+  async handleAPIResponse(responseData: any) {
```
Comment (asaltste) [inline src/Classes/APIBots/ReservoirListBot.ts:38]: nit: Is there not a type we can get from reservoir or axios?

File: src/Classes/APIBots/ReservoirListBot.ts [hunk 3]
```diff
@@ -55,7 +63,7 @@ class ReservoirListBot extends APIPollBot {
    * Reservoir API Spec: https://docs.reservoir.tools/reference/getordersasksv2
    * @param {*} msg - Dict of event data from API response
    */
-  async buildDiscordMessage(msg) {
+  async buildDiscordMessage(msg: any) {
```
Comment (asaltste) [inline src/Classes/APIBots/ReservoirListBot.ts:66]: same question about either getting a type for this or making the type through an enum or interface

File: src/Classes/APIBots/ArchipelagoBot.ts [hunk 4]
```diff
@@ -178,15 +186,15 @@ class ArchipelagoBot {
 
 // Copied from Archipelago UI codebase.
 const priceToString = (
-  price,
-  { decimals = 18, maximumFractionDigits, precisionAdjustment = 0 } = {}
+  price: any,
+  { decimals = 18, maximumFractionDigits, precisionAdjustment = 0 }: any = {}
```
Comment (asaltste) [inline src/Classes/APIBots/ArchipelagoBot.ts:190]: this type seems to be {decimals: number, maximumFractionDigits: number, precisionAdjustment: number}

**Review by asaltste — APPROVED** [review]
This is sweet. Assuming we are going to go back again and fix the any types and then re-add back the running of tests?

File: src/Classes/APIBots/ReservoirListBot.ts (hunk 2 above)
Comment (grantoesterling) [reply src/Classes/APIBots/ReservoirListBot.ts:38]: I was gonna add types for these things when i refactor the API bots a lil more

**Comment by grantoesterling** [comment]
yessir

**Comment by asaltste** [comment]
> yessir

Sweet just wanted to point out those things but if aware all good to 🚢

File: src/Classes/APIBots/ReservoirListBot.ts (hunk 3 above)
Comment (grantoesterling) [reply src/Classes/APIBots/ReservoirListBot.ts:66]: same as the above - will go back and add these types when i refactor these files

**Pull request merged**